spec.html

```diff
@@ -13760,13 +13766,59 @@ <h2>Syntax</h2>
       SingleLineComment ::
         `//` SingleLineCommentChars?
 
+      SingleLineHTMLOpenComment ::
+        `&lt;!--` SingleLineCommentChars?
+
+      SingleLineHTMLCloseComment ::
+        LineTerminatorSequence HTMLCloseComment
+
+      HTMLCloseComment ::
+        WhiteSpaceSequence? SingleLineDelimitedCommentSequence? `--&gt;` SingleLineCommentChars?
+
+      SingleLineDelimitedCommentSequence ::
+        SingleLineDelimitedComment WhiteSpaceSequence? SingleLineDelimitedCommentSequence?
+
+      WhiteSpaceSequence ::
+        WhiteSpace WhiteSpaceSequence?
+
       SingleLineCommentChars ::
         SingleLineCommentChar SingleLineCommentChars?
 
       SingleLineCommentChar ::
         SourceCharacter but not LineTerminator
+
+      SingleLineDelimitedComment ::
+        `/*` SingleLineDelimitedCommentChars? `*/`
+
+      SingleLineDelimitedCommentChars ::
+        SingleLineNotAsteriskChar SingleLineDelimitedCommentChars?
+        `*` SingleLinePostAsteriskCommentChars?
+
+      SingleLineNotAsteriskChar ::
+        SourceCharacter but not one of `*` or LineTerminator
+
+      SingleLinePostAsteriskCommentChars ::
+        SingleLineNotForwardSlashOrAsteriskChar SingleLineDelimitedCommentChars?
+        `*` SingleLinePostAsteriskCommentChars?
+
+      SingleLineNotForwardSlashOrAsteriskChar ::
+        SourceCharacter but not one of `/` or `*` or LineTerminator
     </emu-grammar>
-    <p>A number of productions in this section are given alternative definitions in section <emu-xref href="#sec-html-like-comments"></emu-xref></p>
+
+    <emu-clause id="sec-comments-early-errors">
+      <h1>Static Semantics: Early Errors</h1>
+      <emu-grammar>
+        SingleLineHTMLOpenComment ::
+          `&lt;!--` SingleLineCommentChars?
+
+        HTMLCloseComment ::
+          WhiteSpaceSequence? SingleLineDelimitedCommentSequence? `--&gt;` SingleLineCommentChars?
+      </emu-grammar>
+      <ul>
+        <li>It is a Syntax Error if a |Module| contains the source code matching this production.</li>
+      </ul>
+      <emu-note>In a |Script|, this syntax is allowed, but deprecated.</emu-note>
```

Comment:
It would be nice to clarify in https://tc39.es/ecma262/#sec-syntactic-and-lexical-grammars that "deprecated/legacy" means "optional for implementation".

However, I am in line with @jmdyck about [putting deprecated behavior under something like `if HostImplementsOptional()`](https://github.com/tc39/ecma262/issues/1595#issuecomment-503809775) instead, but without "units" for now.

Reply:
> It would be nice to clarify in https://tc39.es/ecma262/#sec-syntactic-and-lexical-grammars that "deprecated/legacy" means "optional for implementation".

It doesn't mean that. See [here](https://github.com/tc39/ecma262/pull/2125/files#diff-181371b08d71216599b0acccbaabd03c306da6de142ea6275c2135810999805aR154) for a pending definition of "legacy".

> However, I am in line with @jmdyck about [putting deprecated behavior under something like `if HostImplementsOptional()`](https://github.com/tc39/ecma262/issues/1595#issuecomment-503809775) instead, but without "units" for now.

That suggestion is for when you take stuff from Annex B and incorporate it into the main spec, *keeping it normative-optional*. This PR (and #1867) take stuff from Annex B and incorporate it into the main spec, but *make it normative*.